server/src/main/java/com/linecorp/centraldogma/server/metadata/MetadataService.java

```diff
@@ -119,8 +123,26 @@ public MetadataService(ProjectManager projectManager, CommandExecutor executor)
                     continue;
                 }
 
+                final CompletableFuture<Revision> future = new CompletableFuture<>();
+                final CompletableFuture<Revision> futureInMap =
+                        reposInAddingMetadata.computeIfAbsent(repo, key -> future);
```

Comment:
Don't we need to include `projectName` in the cache key?

Reply:
Oops thanks!

Reply:
Fixed. 😉